tket/src/Circuit/CircUtils.cpp

```diff
@@ -105,6 +105,13 @@ Eigen::Matrix4cd get_matrix_from_2qb_circ(const Circuit &circ) {
           }
           break;
         }
+        case OpType::TK2: {
+          auto params = o->get_params();
+          TKET_ASSERT(params.size() == 3);
```

Comment:
I don't fully understand this function, but it seems that the CX code has more complex logic (in terms of `uqb ==  0` and the endianness switch) than the TK2 code. Why is this?